### PR TITLE
Cleaned up retrieval of all previous game data.

### PR DESCRIPTION
**Pull request by nathanieldschneider**
Hey homies, 

Pull request has some merge errors that I will fix tomorrow.

Basically this PR just cleans up the process of pulling a list of all previous statistics. Now it pulls the player information from the database and writes that to a hashmap, and then uses that map to create the correct player objects. It also no longer passes a winnerID.

Also changed it so I'm getting resultset variables by column name and not index. Refactored the FullStats class a bit, Pedro needs to come up with a name because I'm too sleepy. 

Rhys fixed my loop logic problem so that's updated now.

### CODE REVIEW

File: src/main/java/com/toptrumps/db/Statistics.java [hunk 1]
```diff
@@ -35,87 +42,92 @@ private ResultSet retrieveHighStats(){
...
+                    allStats.get(counter).buildRoundStats(roundStats,players);
+                    
                 }else{
-                    counter++;
-                    if(roundStats.getInt(2) == allStats.get(counter).getGameID()){
-                    allStats.get(counter).buildRoundStats(roundStats);
-                    }
+                        counter++;
+                        if(gameId == allStats.get(counter).getGameID()){
+                        allStats.get(counter).buildRoundStats(roundStats,players);
```
Comment (rjs68) [inline src/main/java/com/toptrumps/db/Statistics.java]: I sent you a solution to this boiiii

**Review by rjs68 — CHANGES_REQUESTED** [review]

File: src/main/java/com/toptrumps/db/Statistics.java (hunk 1 above)
Comment (nathanieldschneider) [reply src/main/java/com/toptrumps/db/Statistics.java]: implemented :) well done!

**Review by rjs68 — APPROVED** [review]

File: src/main/java/com/toptrumps/db/FullStats.java [hunk 2]
```diff
@@ -5,48 +5,46 @@
...
-        this.players = players;
-        this.gameID = rs.getInt(1);
-        this.winnerID = rs.getInt(2);
-        this.roundsPlayed = rs.getInt(3);
-        this.draws = rs.getInt(4);
-        this.winner = players.getPlayer(this.winnerID-1);
+    public FullStats(ResultSet rs,Map<Integer,String> players) throws SQLException{
+
+        this.gameID = rs.getInt("game_id");
+        this.winner = new Player(rs.getInt("winner_id"),players.get(rs.getInt("winner_id")));
```
Comment (Arden488) [inline src/main/java/com/toptrumps/db/FullStats.java]: Please-please-please, make three lines for this, instead of one. It's super unreadable

**Review by Arden488 — APPROVED** [review]
I'm going to approve. It's better than it was, but I would suggest reformating code, it's easy to do with IDE such as IntelliJ. Just add spaces and linebreaks

**Comment by Arden488** [comment]
And there are conflicts. Can you fix it by yourself or you need help?

**Review by BelfDev — APPROVED** [review]

**Comment by BelfDev** [comment]
Please try to play a full game after you resolve the conflicts

**Review by hughwin — APPROVED** [review]

**Pull request merged**